scrooge-generator-tests/src/test/resources/gold_file_output_scala/com/twitter/scrooge/test/gold/thriftscala/PlatinumService.scala

```diff
@@ -1083,27 +1083,25 @@ object PlatinumService extends _root_.com.twitter.finagle.thrift.GeneratedThrift
         }
       }
 
-    private def recordRequest(method: ThriftMethod): Unit = {
-      if (perEndpointStats) {
-        val methodStats = ThriftMethodStats((if (serviceName != "") stats.scope(serviceName) else stats).scope(method.name))
-        methodStats.requestsCounter.incr()
+    private def recordRequest(methodStats: Option[ThriftMethodStats]): Unit = {
+      methodStats.foreach { stats =>
+        stats.requestsCounter.incr()
       }
     }
 
-    private def recordResponse(reqRep: ctfs.ReqRep, method: ThriftMethod): Unit = {
+    private def recordResponse(reqRep: ctfs.ReqRep, methodStats: Option[ThriftMethodStats]): Unit = {
       ServerToReqRep.setCtx(reqRep)
-      if (perEndpointStats) {
-        val methodStats = ThriftMethodStats((if (serviceName != "") stats.scope(serviceName) else stats).scope(method.name))
+      methodStats.foreach { stats =>
```

Comment:
similar comment here around avoiding the `Option.foreach`. 

although, actually. here's another approach that i think i would prefer.

instead of using None to represent that case, create a singleton instance of `ThriftMethodStats` that uses a `NullStatsReceiver`. this instance could live on the ThriftMethodStats companion object. or create one as needed, either way.

Reply:
That sounds better to me as well. Before I go ahead and make that change, though, do you think it might make more sense to just bypass calls to these methods entirely in the event that perEndpointStats is false?

Reply:
my preference is the simpler code. that's why i like the "sentinel" object approach leaving just one code path.